### PR TITLE
Return proper errors for exec failure

### PR DESCRIPTION
**Pull request by crosbymichael**
This is only to help debugging the exec errors because they correct
error message is not being returned or is being hidden from the client.

ref: #17473

Signed-off-by: Michael Crosby crosbymichael@gmail.com

### CODE REVIEW

**Comment by tiborvass** [comment]
LGTM

File: api/server/router/container/exec.go [hunk 1]
```diff
@@ -110,10 +110,8 @@ func (s *containerRouter) postContainerExecStart(ctx context.Context, w http.Res
 
 	// Now run the user process in container.
 	if err := s.backend.ContainerExecStart(execName, stdin, stdout, stderr); err != nil {
-		if execStartCheck.Detach {
-			return err
-		}
 		logrus.Errorf("Error running exec in container: %v\n", utils.GetErrorMessage(err))
+		return err
```
Comment (cpuguy83) [inline api/server/router/container/exec.go]: error message above will be duplicated in the daemon logs now, I think.

Comment (calavera) [inline api/server/router/container/exec.go]: yep, we log that error in https://github.com/docker/docker/blob/master/api/server/server.go#L144

Comment (crosbymichael) [inline api/server/router/container/exec.go]: i will remove this once the userns test run is done

**Pull request closed without merging**